### PR TITLE
Add some transforms

### PR DESCRIPTION
**Pull request by marc-moreaux**
In their paper https://arxiv.org/abs/1711.10282, the authors use 2D convolution of monoChannel raw audio sounds with some data augmentation like stretching the sound, randomly cropping a piece of sound and padding the sound on both extremities to allow random crops on the borders

### CODE REVIEW

File: torchaudio/transforms.py [hunk 1]
```diff
@@ -444,3 +452,136 @@ def __call__(self, x_mu):
 
     def __repr__(self):
         return self.__class__.__name__ + '()'
+
+
+class Pad(object):
```
Comment (jamarshon) [inline torchaudio/transforms.py]: This class seems very similar to PadTrim (https://github.com/pytorch/audio/blob/7ee1d389bb07a77e3fcbc4afcfa59fbb96306d04/torchaudio/transforms.py#L75).
Is there any difference?

Comment (Mistobaan) [reply torchaudio/transforms.py]: does not look like. They bot end up using the same `torch.functional.pad` function.

**Comment by Mistobaan** [comment]
Would be nice to merge those extra utility functions @marc-moreaux

File: torchaudio/functional.py [hunk 2]
```diff
@@ -350,3 +356,120 @@ def mu_law_expanding(x_mu, qc):
...
+    Returns:
+        Tensor: a piece of the tensor
+    """
+    orig_size = tensor.size(1 - ch_dim)
+    start = torch.randint(0, orig_size - size, (1,))
+    end = start + size
+    return crop_in_between(tensor, start.item(), end.item(), ch_dim)
+
+
+def stretch(tensor, factor, interpolate, ch_dim):
```
Comment (vincentqb) [inline torchaudio/functional.py]: Is this similar to `Resample` or `phase_vocoder`?

**Comment by vincentqb** [comment]
Since this PR is out of date and contains significant overlap with existing features, I will close this PR. Please feel free to open a new issue to discuss the transforms you would like to add within torchaudio so we can discuss how to proceed.

**Pull request closed without merging**